ethcore/src/externalities.rs

```diff
@@ -262,6 +262,8 @@ impl<'a, T, V> Ext for Externalities<'a, T, V> where T: 'a + Tracer, V: 'a + VMT
 			trace!("Suiciding {} -> {} (xfer: {})", address, refund_address, balance);
 			self.state.transfer_balance(&address, refund_address, &balance);
 		}
+
+		self.tracer.trace_suicide(address, balance, refund_address.clone(), self.depth + 1);
```

Comment:
I don't like that this line is in externalities, but it seems like the cleanest solution of the problem.